tasks/main.yml

```diff
@@ -1,2 +1,44 @@
 ---
-# tasks file for skeleton
+- name: Load var file with package names based on the OS type
+  ansible.builtin.include_vars: "{{ lookup('first_found', params) }}"
+  vars:
+    params:
+      files:
+        - "{{ ansible_distribution }}_{{ ansible_distribution_release }}.yml"
+        - "{{ ansible_distribution }}.yml"
+        - "{{ ansible_os_family }}.yml"
+      paths:
+        - "{{ role_path }}/vars"
+
+- name: Install Samba client
+  ansible.builtin.package:
+    name: "{{ client_packages }}"
+
+- name: Install and enable Samba server
+  block:
+    - name: Install Samba server packages
+      ansible.builtin.package:
+        name: "{{ server_packages }}"
+    - name: Enable Samba service
+      ansible.builtin.service:
+        enabled: yes
+        name: "{{ service_name }}"
+    - name: Create Samba guest user
+      block:
+        - name: Create Samba guest user
+          ansible.builtin.user:
+            comment: Samba guest account
+            create_home: no
+            home: /nonexistent
```

Comment:
Not to pick nits, but since @mcdonnnj isn't here... 

Should `home` vary based on the distro?

See: https://unix.stackexchange.com/a/75329

Reply:
I think as long as it points to a place that doesn't exist it's fine.  Would you prefer `/dev/null`?

Reply:
I would prefer `/dev/null`.  That is what I've always used.  It seems the world has changed.  As long as this isn't a new "bad practice" I think it makes more sense to point to something that is guaranteed not to give a user access.  I could see a configuration like this being part of a hacking challenge.  

I feel heard.  Please feel free to resolve this conversation pending any action or inaction you see fit.

Reply:
Please see commit 935268e.